### PR TITLE
Make sure current task build id works with queries / caching

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Made the current task build id sent on activations consistent with what I just did in Go / Java.

## Why?
Part of https://github.com/temporalio/features/issues/253

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: core/src/worker/workflow/machines/workflow_machines.rs [hunk 1]
```diff
@@ -420,9 +418,15 @@ impl WorkflowMachines {
                 Some(workflow_activation_job::Variant::QueryWorkflow(_))
             )
         });
+        let is_replaying = self.replaying || all_query;
+        let build_id_for_current_task = if is_replaying {
+            self.current_wft_build_id.clone().unwrap_or_default()
```
Comment (Quinn-With-Two-Ns) [inline core/src/worker/workflow/machines/workflow_machines.rs:423]: Does core need the same logic as Go to check for an empty build ID?

File: core/src/worker/workflow/machines/workflow_machines.rs [hunk 2]
```diff
@@ -420,9 +418,15 @@ impl WorkflowMachines {
                 Some(workflow_activation_job::Variant::QueryWorkflow(_))
             )
         });
+        let is_replaying = self.replaying || all_query;
```
Comment (Quinn-With-Two-Ns) [inline core/src/worker/workflow/machines/workflow_machines.rs:421]: Aren't queries considered replaying anyway?

Comment (Sushisource) [reply core/src/worker/workflow/machines/workflow_machines.rs:421]: Not until this point, that's where it happens

File: core/src/worker/workflow/machines/workflow_machines.rs (hunk 1 above)
Comment (Sushisource) [reply core/src/worker/workflow/machines/workflow_machines.rs:423]: The lookahead works a bit differently and avoids this problem - I actually ended up changing the Go one to be a bit simpler and more similar in my last fix too.

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Pull request merged**